apis/Google.Cloud.PubSub.V1/docs/index.md

```diff
@@ -82,6 +82,139 @@ restart listening for messages with `StartAsync(...)` again. Due to the expense
 application.
 
+## Dependency Injection
+
+Both `PublisherClient` and `SubscriberClient` can be easily integrated with the dependency injection
+container provided by the [Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection/) package.
+
+### PublisherClient
+
+To register a singleton `PublisherClient` instance with default settings in the `IServiceCollection`, use the
+`AddPublisherClient` extension method as shown below:
+
+```csharp
+services.AddPublisherClient(TopicName.FromProjectTopic("projectId", "topicId"));
+```
+
+There is an overload of `AddPublisherClient` method that takes `Action<PublisherClientBuilder>` as a parameter
+and can be used to add the customized `PublisherClient` singleton instance as shown below:
+
+```csharp
+services.AddPublisherClient(builder =>
+{
+    builder.TopicName = TopicName.FromProjectTopic("projectId", "topicId");
+    builder.CredentialsPath = "path/to/credentials.json";
+    // Other settings to customize the client.
+});
+```
+
+Using the `PublisherClient` registered with the dependency injection container is easy. To do so, simply create a constructor for the class in which the `PublisherClient`
+is required to be used and add the `PublisherClient` as a parameter.
+
+For example:
+
+```csharp
+public class MyController : ControllerBase
+{
+    private readonly PublisherClient _publisherClient;
+
+    public MyController(PublisherClient publisherClient)
+    {
+        _publisherClient = publisherClient;
+    }
+
+    // Use the _publisherClient to publish messages in a method.
+}
+```
+
+When the application exits, the `DisposeAsync` method of the `PublisherClient` will be invoked automatically to gracefully shut down the client.
+
+### SubscriberClient
+
+To register a singleton `SubscriberClient` instance with default settings in the `IServiceCollection`, use the
+`AddSubscriberClient` extension method as shown below:
+
+```csharp
+services.AddSubscriberClient(SubscriptionName.FromProjectSubscription("projectId", "subscriptionId"));
+```
+
+There is an overload of `AddSubscriberClient` method that takes `Action<SubscriberClientBuilder>` as a parameter
+and can be used to add the customized `SubscriberClient` singleton instance as shown below:
+
+```csharp
+services.AddSubscriberClient(builder =>
+{
+    builder.SubscriptionName = SubscriptionName.FromProjectSubscription("projectId", "subscriptionId");
+    builder.CredentialsPath = "path/to/credentials.json";
+    // Other settings to customize the client.
+});
+```
+
+The `SubscriberClient` is a long-running client and so it may be useful to use
+it in a background service. The background service can use the `SubscriberClient`
+registered with the dependency injection container and handle the messages in the background.
+
+In ASP.NET Core, the background services can be registered with the dependency injection container
+using the `AddHostedService` extension method as shown below:
```

Comment:
What package/namespace is this on? Use full names the first time you mention a type, please.

Reply:
Again, mentioning the origin of a method is always useful. If it's weird to se the full name for an extension method (I agree it is), do something like (`AddHostedService`)[link to MS reference doc for this method].

Time and time again, we (across all languages) have received feedback from users that imports are not clear in samples.

Reply:
Sure Amanda. I will do so. Thanks.